c7n/resources/sagemaker.py

```diff
@@ -333,13 +382,8 @@ class StopNotebookInstance(BaseAction, StateTransitionFilter):
     def process_instance(self, resource):
         client = local_session(
             self.manager.session_factory).client('sagemaker')
-        try:
-            client.stop_notebook_instance(
```

Comment:
just to be clear, its not that we want to remove exception handling, its that we want to catch only known exceptions that are handable. ie. ResourceNotFound we would catch, but AccessDenied we wouldn't.

Reply:
Right. You said to check the API doc on common service errors... the only service I could find in SageMaker that clearly defined ResourceNotFound for any of the specific resources was the training-job (outside of the Ruby api docs, anyways)